### PR TITLE
feat: Log in using the auth dApp

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR changes the way the `UserInformation` component is built by adding the authorization function which redirects the user to the authorization site.

### CODE REVIEW

File: src/hooks/useAuth.ts [hunk 1]
```diff
@@ -26,9 +26,20 @@ export { initialState }
...
+  {
+    authPath,
+  }: {
+    authPath: string
+  } = { authPath: '/auth' }
+) {
   const [state, setState] = useState<AuthState>({ ...initialState })
 
+  const authorize = useCallback(() => {
+    window.location.replace(`${authPath}?redirectTo=${window.location.href}`)
```
Comment (meelrossi) [inline src/hooks/useAuth.ts:39]: should we add `/login` here?

File: src/hooks/useAuth.utils.ts [hunk 2]
```diff
@@ -102,7 +102,10 @@ export async function restoreConnection(): Promise<AuthState> {
 
       const currentAccounts = await fetchAccounts(provider)
       const account = currentAccounts[0]
-      const identity = await SSO.getIdentity(account)
+      // Get the identity first from the local storage. If it doesn't exist, get it from the iframe.
+      const identity =
+        (await SSO.localStorageGetIdentity(account)) ??
```
Comment (meelrossi) [inline src/hooks/useAuth.utils.ts:107]: is the first one async?

File: src/hooks/useAuth.utils.ts [hunk 3]
```diff
@@ -147,7 +150,7 @@ export async function createConnection(
   chainId: ChainId
 ) {
   try {
-    const data = await connection.connect(providerType, chainId)
+    const data = await connection.connect(providerType as any, chainId)
```
Comment (meelrossi) [inline src/hooks/useAuth.utils.ts:153]: why is this as any needed?

**Review by meelrossi — APPROVED** [review]

File: src/hooks/useAuth.utils.ts (hunk 2 above)
Comment (LautaroPetaccio) [reply src/hooks/useAuth.utils.ts]: It's not! I've removed it.

File: src/hooks/useAuth.utils.ts (hunk 3 above)
Comment (LautaroPetaccio) [reply src/hooks/useAuth.utils.ts]: There was a development issue that was later not needed, I've removed it.

File: src/hooks/useAuth.ts (hunk 1 above)
Comment (LautaroPetaccio) [reply src/hooks/useAuth.ts]: We do! Removed it

**Pull request merged**